tests/common/util.rs

```diff
@@ -1189,14 +1239,284 @@ impl UCommand {
+    }
+
+    /// Reads and returns all bytes from child's stderr. Do not call this method if the child process is
+    /// still running. If stderr is redirected to stdout with [`UCommand::stderr_to_stdout`] then always
+    /// zero bytes are returned.
+    ///
+    /// # Important
+    /// This method will block indefinitely if the child process is still running
+    pub fn stderr_bytes(&mut self) -> Vec<u8> {
+        let mut buffer = Vec::<u8>::new();
+        if self.output.is_none() {
+            let stderr = self.raw.stderr.as_mut().unwrap();
+            stderr.read_to_end(&mut buffer).unwrap();
+        }
+
+        buffer
+    }
+
+    /// Reads and returns the exact amount of bytes from stderr. If stderr is redirect to stdout with
+    /// [`UCommand::stderr_to_stdout`] then always zero bytes are returned.
+    ///
+    /// # Important
+    /// This method blocks indefinitely if the `size` amount of bytes cannot be read.
+    pub fn stderr_exact_bytes(&mut self, size: usize) -> Vec<u8> {
+        match &self.output {
+            Some(_) => vec![],
+            None => {
+                let mut buffer: Vec<u8> = vec![0; size];
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                stderr.read_exact(&mut buffer).unwrap();
+                buffer
+            }
+        }
+    }
+
+    /// Returns true if the child process is still alive and false otherwise.
+    ///
+    /// # Panics
+    /// If an error occurred during try_wait()
+    pub fn try_alive(&mut self) -> bool {
+        self.raw.try_wait().unwrap().is_none()
+    }
+
+    // TODO: improve this method and recognize if the the command already exited and if so return
+    // the code and success from the command. And maybe panic! if the process is still running and
+    // not run together with stderr_to_stdout, since this method will block indefinitely otherwise.
+    pub fn assert(&mut self) -> CmdResult {
+        CmdResult {
+            bin_path: self.bin_path.clone(),
+            util_name: self.util_name.clone(),
+            tmpd: self.tmpd.clone(),
+            code: None,
+            success: true, // TODO: or is false better ?
+            stdout: self.stdout_bytes(),
+            stderr: self.stderr_bytes(),
+        }
+    }
+
+    // TODO: rename to assert_between ?
+    /// Collects the current output of the child process. It consumes from stdout the amount of
+    /// bytes given by `expected_stdout_size` and from stderr the `expected_stderr_size`. If you've
+    /// redirected stderr to stdout with `stderr_to_stdout` then no bytes are consumed. Returns a
+    /// [`CmdResult`] on which all the usual assertions can be done.
+    ///
+    /// Note, that [`CmdResult::code`] and associated methods will panic. Since the process is still
+    /// running there is no return code to assert. Methods like [`CmdResult::success`] will always
+    /// evaluate to true.
+    pub fn assert_now(
```

Comment:
Maybe it could be called `intermediate_result`? Ideally this method would be unnecessary if the other methods aren't blocking and return intermediate stdout etc.

Reply:
What I really like about this assert prefix is how the tests read: `child.assert_now().stdout_only()...`. Since we're writing within a test framework, I think we can use such naming of methods to keep tests more readable. What about `assert_that(..)`?

Reply:
I understand the idea, but in my opinion that connection is not obvious enough. I would prefer a more descriptive name.

Reply:
I honestly don't like name discussions too much, and read is not describing what you're actually doing.

What's more descriptive than (with regard to remove these expected... arguments)
```rust
child.assert_now_that().stdout_only()
``` 

and you're asserting something about the child? I think, the `that` in the name clearly indicates you're expected to assert something after the call to this method.

Reply:
> I honestly don't like name discussions too much

Same, let's leave it and come back to this once the rest of the API is more clear and whether we actually need this.

Reply:
Ok